src/ess/nmx/mcstas_loader.py

```diff
@@ -29,26 +31,33 @@
 """A function that derives arbitrary proton charge based on event weights."""
 
 
-def _retrieve_event_list_name(keys: Iterable[str]) -> str:
-    prefix = "bank01_events_dat_list"
+def _retrieve_event_list_names(keys: Iterable[str]) -> tuple[str, ...]:
+    import re
 
-    # (weight, x, y, n, pixel id, time of arrival)
     mandatory_fields = 'p_x_y_n_id_t'
+    # (weight, x, y, n, pixel id, time of arrival)
+    pattern = r"bank(\d+\d+)_events_dat_list_" + mandatory_fields
```

Comment:
```suggestion
    pattern = r"bank(\d\d+)_events_dat_list_" + mandatory_fields
```
Is this what you mean? 2 or more digits?

Reply:
Aaaah yes! thanks : D